pkg/eventhandler/eventhandler.go

```diff
@@ -50,28 +51,36 @@ func (n *EventHandler) HandleGrantEvent(ctx context.Context, log *zap.SugaredLog
 	if err != nil {
 		return err
 	}
+	// This would indicate a race condition or a major error
 	if gq.Result.Grant == nil {
 		return fmt.Errorf("request: %s does not have a grant", grantEvent.Grant.ID)
 	}
-	oldStatus := gq.Result.Status
-	switch event.DetailType {
-	case gevent.GrantActivatedType:
-		gq.Result.Grant.Status = ac_types.ACTIVE
-	case gevent.GrantExpiredType:
-		gq.Result.Grant.Status = ac_types.EXPIRED
-	case gevent.GrantFailedType:
-		gq.Result.Grant.Status = ac_types.ERROR
-	// revoking is handling as a synchronous operation, so we do not modify the database for these events as it is handled in the API already
-	// case gevent.GrantRevokedType:
-	// 	gq.Result.Status = ac_types.GrantStatusREVOKED
-	default:
-		zap.S().Infow("unhandled grant event type", "detailType", event.DetailType)
-	}
-	log.Infow("updating grant status on request", "old", oldStatus, "new", gq.Result.Status)
-	items, err := dbupdate.GetUpdateRequestItems(ctx, n.db, *gq.Result)
-	if err != nil {
-		return err
+	oldStatus := gq.Result.Grant.Status
+	newStatus := grantEvent.Grant.Status
+	gq.Result.Grant.Status = newStatus
+
+	// don't add status changed f event is its the same status
+	// but check if its grant created
+	if oldStatus == newStatus {
+		// I anticipate that this would be succeptible to a race condition, recoverable if the eventbridge retries the event handler
+		// this is because the grant events are sourced from the access handler prior to the request being saved to dynamodb on creation
+		// we could solve this by saving the request to the DB prior to making the call to the access handler?
+		if event.DetailType == gevent.GrantCreatedType {
+			requestEvent := access.NewGrantCreatedEvent(gq.Result.ID, time.Now(), nil)
+			log.Infow("inserting request event for grant created")
+			return n.db.Put(ctx, &requestEvent)
+		}
+		return nil
```

Comment:
I found this a bit confusing to read - we could make it simpler by reducing the levels of nesting

ie

```
if oldStatus != newStatus {
  // emit status change event
}

if event.DetailType == gevent.GrantCreatedType {
  // emit grant created event
}
```